vmware/ngc/NGC-Plugin/src/main/java/org/opensds/vmware/ngc/controller/DeviceController.java

```diff
@@ -0,0 +1,124 @@
+// under the License.
+
+package org.opensds.vmware.ngc.controller;
+
+import com.google.gson.Gson;
+import org.opensds.vmware.ngc.entity.ResultInfo;
+import org.opensds.vmware.ngc.model.DeviceInfo;
+import org.opensds.vmware.ngc.service.DeviceService;
+import org.opensds.vmware.ngc.util.ObjectIdUtil;
+import org.opensds.vmware.ngc.util.QueryUtil;
+import com.vmware.vise.vim.data.VimObjectReferenceService;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+
+@Controller
+@RequestMapping(value = "/device")
+
+public class DeviceController  {
+    private static final Log logger = LogFactory.getLog(DeviceController.class);
+    @Autowired(required=false)
+    private DeviceService deviceService;
+    @Autowired
+    private VimObjectReferenceService objectReferenceService;
+
+
+    public DeviceController() {
+        QueryUtil.setObjectReferenceService(objectReferenceService);
+    }
+
+    /**
+     * @param json
+     * @return
+     */
+    @RequestMapping(value = "/add", method = RequestMethod.POST)
+    @ResponseBody
+    public ResultInfo add(
+            @RequestParam(value = "json") String json) {
+        Gson gson = new Gson();
+        DeviceInfo deviceInfo = gson.fromJson(json, DeviceInfo.class);
+        ResultInfo resultInfo = deviceService.add(deviceInfo);
+        return resultInfo;
+    }
+
+    /**
+     * @param deviceID
+     * @return
+     */
+    @RequestMapping(value = "/delete", method = RequestMethod.DELETE)
+    @ResponseBody
+    public ResultInfo delete(
+            @RequestParam(value = "deviceID") String deviceID) {
+        Object objectReference = getObjectReference(deviceID);
+        ResultInfo result = deviceService.delete(objectReference);
+        return result;
+    }
+
+    @RequestMapping(value = "/get", method = RequestMethod.GET)
+    @ResponseBody
+    public ResultInfo get(
+            @RequestParam(value = "deviceID") String deviceID) {
+        Object objectReference = getObjectReference(deviceID);
+        ResultInfo resultInfo = deviceService.get(objectReference);
+        return resultInfo;
+    }
+
+    /**
+     * @param deviceID
+     * @param json
+     * @return
+     */
+    @RequestMapping(value = "/update", method = RequestMethod.PUT)
+    @ResponseBody
+    public ResultInfo update(
+            @RequestParam(value = "deviceID") String deviceID,
+            @RequestParam(value = "json") String json) {
+        Gson gson = new Gson();
+        DeviceInfo deviceInfo = gson.fromJson(json, DeviceInfo.class);
```

Comment:
@kumarashit According to GSon documentation an exception is already thrown if the json stream can't be deserialized 
It will throws:  JsonParseException - if json is not a valid representation for an object of that class

Reply:
My point was referencing the error handling and exception handling in context by callee than the caller doing it. 
Ok for me. You can mark this as resolved